src/tiled/templategroupdocument.cpp

```diff
@@ -94,3 +102,125 @@ void TemplateGroupDocument::addTemplate(ObjectTemplate *objectTemplate)
 {
     mTemplateGroup->addTemplate(objectTemplate);
 }
+
+
+static void writeTemplateDocumentsXml(QFileDevice *device,
+                                      const TemplateDocuments &templateDocuments)
+{
+    QXmlStreamWriter writer(device);
+
+    writer.setAutoFormatting(true);
+    writer.setAutoFormattingIndent(1);
+
+    writer.writeStartDocument();
+    writer.writeStartElement(QLatin1String("templateDocuments"));
+
+    for (const TemplateGroupDocument *templateDocument : templateDocuments) {
+        writer.writeStartElement(QLatin1String("templatedocument"));
+
+        writer.writeAttribute(QLatin1String("path"), templateDocument->fileName());
+
+        writer.writeEndElement();
+    }
+
+    writer.writeEndElement();
+    writer.writeEndDocument();
+}
+
+static void readTemplateDocumentsXml(QFileDevice *device,
+                                     TemplateDocuments &templateDocuments,
+                                     QString &error)
+{
+    QXmlStreamReader reader(device);
+
+    if (!reader.readNextStartElement() || reader.name() != QLatin1String("templatedocuments")) {
+        error = QCoreApplication::translate(
+                    "TemplateDocuments", "File doesn't contain template documents.");
+        return;
+    }
+
+    while (reader.readNextStartElement()) {
+        if (reader.name() == QLatin1String("templatedocument")) {
+            const QXmlStreamAttributes atts = reader.attributes();
+            const QString path(atts.value(QLatin1String("path")).toString());
+
+            auto templateGroupFormat = new TtxTemplateGroupFormat();
+
+            // TODO: handle errors that might happen while loading
```

Comment:
Right, at least don't put `nullptr` into `templateDocuments`. :-)

Also, don't forget to initialize `templateGroup` in `MapReaderPrivate::readTemplateGroup` with `nullptr` or you're going to have trouble detecting errors.